cmd/argocd/commands/project.go

```diff
@@ -618,10 +618,44 @@ func modifyResourcesList(list *[]metav1.GroupKind, add bool, listDesc string, gr
 	return true
 }
 
+func modifyResourcesDenyList(list *[]v1alpha1.BlacklistEntry, add bool, visible bool, listDesc string, group string, kind string) bool {
```

Comment:
passing a pointer of an array and modifying it inside an unexported function is not a very good pattern - it actually hides state modification - I highly recommend to change the function to accept a list and then return the modified list